### PR TITLE
Trim react-native from the name in the same places create-rn-library and Codegen does

### LINKED ISSUE #109: Generated code is being prefixed (e.g. with "ReactNative") but some occurrences appear to be missing
I have the following in `package.json`:

```json5
"codegenConfig": {
  "name": "RNMatrixSdkSpec",
  "type": "modules",
  "jsSrcsDir": "src"
}
```

When generating the turbo module with `46f81224e4aae9a05dc7c8e209b4750c014f4d02` (main branch from an hour ago), the iOS module comes out as

```
// Generated by uniffi-bindgen-react-native
#ifdef __cplusplus
#import "react-native-matrix-sdk.h"
#endif

#ifdef RCT_NEW_ARCH_ENABLED
#import "RNMatrixSdkSpec.h"

@interface ReactNativeMatrixSdk : NSObject <RNMatrixSdkSpec>
#else
#import <React/RCTBridgeModule.h>

@interface ReactNativeMatrixSdk : NSObject <RCTBridgeModule>
#endif

@end
```

Notice how it imports `RNMatrixSdkSpec.h` and tries to implement the eponymous protocol `<RNMatrixSdkSpec>`. However, the generated file in `example/ios/build/generated/ios/RNMatrixSdkSpec/RNMatrixSdkSpec.h` defines the protocol as `@protocol NativeReactNativeMatrixSdkSpec`. As a result, the build on iOS fails.

When I change the name in `package.json` to `NativeReactNativeMatrixSdkSpec`

```json5
"codegenConfig": {
  "name": "NativeReactNativeMatrixSdkSpec",
  "type": "modules",
  "jsSrcsDir": "src"
}
```

both the generated file and the protocol come out as `NativeReactNativeMatrixSdkSpec` and the build succeeds.

### PR DESCRIPTION
**Pull request by jhugman**
Fixes #109.

This PR adds testing around the turbo-modules.

The tests runs through a host of different names, then 

- runs `create-react-native-library` with that name. 
- runs the `ubrn generate turbo-module` command
- the checks that generated files from the ubrn command still uses the identifiers in the `create-react-native-library` files
- finally builds for ios and android the library.

This has meant a series of changes (for the better) to the templates.

### CODE REVIEW

**Comment by jhugman** [comment]
I haven't yet got the test running in CI, but I think that can wait for another PR. Marking ready for review.

File: crates/ubrn_cli/src/codegen/mod.rs [hunk 1]
```diff
@@ -264,7 +264,7 @@ mod files {
     templated_file!(IndexTs, "index.ts");
```
Comment (Johennes) [inline crates/ubrn_cli/src/codegen/mod.rs:264]: Should this also end in `.tsx`?

Comment (jhugman) [reply crates/ubrn_cli/src/codegen/mod.rs]: Yes, you're right. I renamed the template file and made this `index.tsx` to match. I also renamed the `IndexTs` struct.

Thanks for prompting me out of laziness. :)

**Review by zzorba — APPROVED** [review]
LGTM

**Pull request merged**